c2corg_api/views/route_tag.py

```diff
@@ -0,0 +1,128 @@
+import logging
+
+from c2corg_api import DBSession
+from c2corg_api.models.route import Route
+from c2corg_api.models.tag import RouteTag
+from c2corg_api.views import cors_policy, restricted_json_view
+from colander import MappingSchema, SchemaNode, Integer, required
+from cornice.resource import resource
+from cornice.validators import colander_body_validator
+
+log = logging.getLogger(__name__)
+
+
+class TagSchema(MappingSchema):
+    route_id = SchemaNode(Integer(), missing=required)
+
+
+def get_tag_relation(user_id, route_id):
+    return DBSession. \
+        query(RouteTag). \
+        filter(RouteTag.user_id == user_id). \
+        filter(RouteTag.route_id == route_id). \
+        first()
+
+def validate_route_id(request, **kwargs):
+    """ Check that the route exists.
+    """
+    route_id = request.validated['route_id']
+    route_exists_query = DBSession.query(Route). \
+        filter(Route.document_id == route_id). \
+        exists()
+    route_exists = DBSession.query(route_exists_query).scalar()
+
+    if not route_exists:
+        request.errors.add(
+            'body', 'route_id', 'route {0} does not exist'.format(route_id))
+
+
+@resource(path='/routes/tag', cors_policy=cors_policy)
+class RouteTagRest(object):
```

Comment:
Suggestion : IMO, having a unique REST service, leveraging on REST model would be better : 

* `GET /routes/tag/{route_id}`=> returns `{"todo": True/False}`
* `PUT /routes/tag/{route_id}`=> , empty body as now, set the tag on (use PUT, not POST, as it's idempotent)
* `DELETE /routes/tag/{route_id}`=> , empty body as now, set the tag off

Reply:
@cbeauchesne Your suggestion makes sense but
* this code is inspired from https://github.com/c2corg/v6_api/blob/master/c2corg_api/views/user_follow.py which itself uses the same approach than other services
* I don't remember exactly but I think it is related to the way Cornice works

As a result I prefer to stick to this approach in order to keep code consistency and to avoid spending too much time using another approach whereas the existing one has proved it worked.